server/protocols/jmap/src/test/java/org/apache/james/jmap/model/KeywordTest.java

```diff
@@ -0,0 +1,289 @@
+        Flags expectedFlags = FlagsBuilder.builder()
+            .add(Flags.Flag.DRAFT)
+            .build();
+        Flags imapFlags = Keyword.toFlagsWithFilterUnsupportedKeywords(ImmutableSet.of(Keyword.DRAFT, Keyword.DELETED));
+
+        assertThat(imapFlags).isEqualToComparingFieldByField(expectedFlags);
+    }
+
+    @Test
+    public void fromKeywordsShouldBuildImapFlagWithUserFlagFromJmapFlags() throws Exception {
+        Flags expectedFlags = FlagsBuilder.builder()
+            .add(Flags.Flag.DRAFT)
+            .add(FORWARDED)
+            .build();
+        Flags imapFlags = Keyword.toFlagsWithFilterUnsupportedKeywords(ImmutableSet.of(Keyword.DRAFT, new Keyword(FORWARDED, true)));
+
+        assertThat(imapFlags).isEqualToComparingFieldByField(expectedFlags);
+    }
+
+    @Test
+    public void fromKeywordsShouldReturnFlagsFromSetOfStringOfKeyword() throws Exception {
+        Flags expected = FlagsBuilder.builder()
+                .add(Flags.Flag.FLAGGED)
+                .add(FORWARDED)
+                .build();
+        assertThat(Keyword.toFlagsWithFilterUnsupportedKeywords(ImmutableSet.of(Keyword.FLAGGED, new Keyword(FORWARDED, true))))
+            .isEqualTo(expected);
+    }
+
+    @Test
+    public void toFlagsShouldReturnFlagsFromListOfStringOfKeyword() throws Exception {
+        Flags expected = FlagsBuilder.builder()
+                .add(Flags.Flag.FLAGGED)
+                .add(FORWARDED)
+                .build();
+        assertThat(Keyword.toFlags("$Flagged", FORWARDED))
+                .isEqualTo(expected);
+    }
+
+    @Test
+    public void buildKeywordsReturnEmptyWhenEmpty() throws Exception {
+        assertThat(Keyword.toSetOfKeywords(Optional.empty())).isEmpty();
+    }
+
+    @Test
+    public void buildKeywordsShouldThrowWhenWrongFlagValue() throws Exception {
+        expectedException.expect(IllegalArgumentException.class);
+
+        Keyword.toSetOfKeywords(Optional.of(ImmutableMap.of("AnyKey", false)));
+    }
+
+    @Test
+    public void buildKeywordsShouldThrowWhenWrongFlagName() throws Exception {
+        expectedException.expect(IllegalArgumentException.class);
+
+        Keyword.toSetOfKeywords(Optional.of(ImmutableMap.of("Any Key", true)));
+    }
+
+    @Test
+    public void buildKeywordsShouldReturnSetOfKeywords() throws Exception {
+        Optional<ImmutableSet<Keyword>> keywords = Keyword.toSetOfKeywords(
+            Optional.of(
+                ImmutableMap.of(
+                    "AnyKey", true,
+                    "OtherKey", true)));
+
+        assertThat(keywords).isPresent();
+        assertThat(keywords.get()).containsOnly(new Keyword("AnyKey", true), new Keyword("OtherKey", true));
+    }
+
+    @Test
+    public void asFlagShouldReturnSystemFlag() throws Exception {
+        assertThat(new Keyword("$Draft", true).asFlag())
+            .isEqualTo(Optional.of(Flags.Flag.DRAFT));
```

Comment:
assertThat of guava does not support for java.util.Optional. Could we accept this?